app/actors/curation_concerns/actors/add_to_collection_actor.rb

```diff
@@ -22,8 +22,11 @@ def add_to_collections(new_collection_ids)
           # remove from old collections
           (curation_concern.in_collection_ids - new_collection_ids).each do |old_id|
             collection = ::Collection.find(old_id)
-            collection.members.delete(curation_concern)
-            collection.save
+            # don't remove from collections current user doesn't have edit permissions for
+            if collection.edit_users.include? user.email
```

Comment:
my instinct is to say that is outside the scope of this PR. Before this PR, works were being deleted from collections as a side-effect of another user saving the work. This fixes that behavior. If certain users need fine-grained deletion abilities, I propose that should be dealt with at another time.